sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkClientOptions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// The options for <see cref="SparkBatchClient"/> and <see cref="SparkSessionClient"/>.
+    /// </summary>
+    public class SparkClientOptions : ClientOptions
+    {
+        private const ServiceVersion Latest = ServiceVersion.V2019_11_01_preview;
+
+        internal static SparkClientOptions Default { get; } = new SparkClientOptions();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SparkClientOptions"/>.
+        /// </summary>
+        public SparkClientOptions(ServiceVersion serviceVersion = Latest)
+        {
+            VersionString = serviceVersion switch
+            {
+                ServiceVersion.V2019_11_01_preview => "2019-11-01-preview",
+                _ => throw new ArgumentOutOfRangeException(nameof(serviceVersion))
+            };
+        }
+
+        /// <summary>
+        /// API version for Spark job service.
+        /// </summary>
+        public string VersionString { get; set; }
```

Comment:
Sorry I might not get your point. The reason I want to make it public here is because I want to give users ability to set API version. It is useful in some scenarios, for example, when we can easily test a new API version in this way.

Reply:
Here's the relevant bit from our [C# Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-versioning-select-api-version)

> ✅ DO allow the consumer to explicitly select a supported service API version when instantiating the service client.
> 
> Use a constructor parameter called version on the client options type.
> 
> - The version parameter must be the first parameter to all constructor overloads.
> - The version parameter must be required, and default to the latest supported service version.
> - The type of the version parameter must be ServiceVersion; an enum nested in the options type.
> - The ServiceVersion enum must use explicit values starting from 1.
> - ServiceVersion enum value 0 is reserved. When 0 is passed into APIs, ArgumentException should be thrown

For our Track 2 clients, we create `ServiceVersion` enums with all the different service versions supported by a client library.  Customers can then select the specific service version using that enum when they create their `ClientOptions` or leave it empty to use the latest.  It's easier than going to the REST docs to find the list of available version strings and it tells them exactly which subset of those versions are supported by this client library.

Generally if you need to do this for testing, we'd suggest a pipeline policy rather than a public API.  [Here's a quick example of using a pipeline policy to change your API version.](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample10_RunningWithDifferentStorageVersion.cs#L187)

I think it's fine to merge your changes in and address this separately after.

Reply:
Hi @tg-msft , we do have a `ServiceVersion` enums . For example: in this file it is defined [here](https://github.com/idear1203/azure-sdk-for-net/blob/synapse_track_2_data_plane_sdk/sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkClientOptions.cs#L38-L46).

According to the discussion, since it is possible to change API version for testing, I will change it to `ServiceVersion` and remove the setter.